src/wayland/dmabuf/mod.rs

```diff
@@ -422,9 +422,23 @@ impl DmabufFeedbackBuilder {
 /// Feedback for dmabuf allocation
 ///
 /// Use the [`DmabufFeedbackBuilder`] to create a new instance.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone)]
 pub struct DmabufFeedback(Arc<DmabufFeedbackInner>);
 
+impl PartialEq for DmabufFeedback {
+    fn eq(&self, other: &Self) -> bool {
+        // Note: The dmabuf feedback can not change, it is initialized
+        // with the DmabufFeedbackBuilder, so if the arc ptr equal we
+        // can short-circuit the expensive equality test saving us
+        // a few cpu cycles
+        if Arc::ptr_eq(&self.0, &other.0) {
+            return true;
+        }
+
+        self.0 == other.0
```

Comment:
Wouldn't the inverse apply? ie, instead of this, why not just `false`

Reply:
No, not necessarily. The feedback could still hold the same formats and tranches, in which case we do not want to send it to the client. Sending the same feedback could result in unnecessary re-allocations in the client.